File: ai/ai-best-practices/openai-hardcoded-api-key/openai-hardcoded-api-key.rb
# ruleid: openai-hardcoded-api-key-ruby
client = OpenAI::Client.new(access_token: "sk-1234567890abcdef")

# ok: openai-hardcoded-api-key-ruby
client = OpenAI::Client.new(access_token: ENV["OPENAI_API_KEY"])

# ok: openai-hardcoded-api-key-ruby
client = OpenAI::Client.new(access_token: get_secret("openai"))

# ok: openai-hardcoded-api-key-ruby
client = OpenAI::Client.new(access_token: "not-a-real-key")

File: ai/ai-best-practices/openai-hardcoded-api-key/openai-hardcoded-api-key.go
package main

import "github.com/sashabaranov/go-openai"

func main() {
	// ruleid: openai-hardcoded-api-key-go
	client := openai.NewClient("sk-1234567890abcdef")

	// ok: openai-hardcoded-api-key-go
	client := openai.NewClient(os.Getenv("OPENAI_API_KEY"))

	// ok: openai-hardcoded-api-key-go
	client := openai.NewClient(apiKey)

	// ok: openai-hardcoded-api-key-go
	client := openai.NewClient("not-a-real-key")
}
